src/tools/utils.rs

```diff
@@ -115,3 +119,53 @@ where
     );
     Ok(())
 }
+
+/// block on function
+pub fn block_on<T, F: Future<Output = T>>(f: F) -> Result<T> {
+    Ok(tokio::runtime::Runtime::new()?.block_on(f))
```

Comment:
I think, it should try to use the current runtime if available. If not - then create a new one